src/ogdc_runner/recipe/viz_workflow.py

```diff
@@ -0,0 +1,319 @@
+        )
+        print_log(f"Writing chunk {idx} to {output_fp}")
+        output_fp.parent.mkdir(parents=True, exist_ok=True)
+        gdf[start : start + num_features].to_file(
+            filename=output_fp,
+            driver="GPKG",
+        )
+        results.append(str(output_fp))
+
+    # Output only the JSON to stdout
+    print(json.dumps(results))
+
+
+@script(
+    name="tiling",
+    inputs=[
+        Parameter(name="chunk-filepath"),
+    ],
+    image="ghcr.io/rushirajnenuji/viz-staging:latest",
+    command=["python"],
+    volume_mounts=[
+        VolumeMount(name="qgnet-ogdc-workflow-pvc", mount_path="/mnt/workflow")
+    ],
+)
+def tiling_process() -> None:
+    """Creates tiles from a geospatial data chunk."""
+    import json
+    import sys
+    from pathlib import Path
+
+    from pdgstaging import TileStager  # type: ignore[import-not-found]
+
+    # Log to stderr
+    def print_log(message: str) -> None:
+        print(message, file=sys.stderr)
+
+    # Read the viz-config.json from the PVC
+    workflow_config = json.loads(
+        Path("/mnt/workflow/config/viz-config.json").read_text()
+    )
+    stager = TileStager(workflow_config, check_footprints=False)
+    print_log("Staging chunk file")
+    print_log("{{inputs.parameters.chunk-filepath}}")
+    stager.stage("{{inputs.parameters.chunk-filepath}}")
+    print_log("Staging done")
+
+
+def make_and_submit_viz_workflow(
+    recipe_config: RecipeConfig,
+    wait: bool,
+    custom_image: str | None = None,
+    custom_tag: str | None = None,
+    custom_namespace: str | None = None,
+    update_global: bool = False,
+    enable_tiling: bool = False,
+    enable_rasterize: bool = False,
+    enable_3dtiles: bool = False,
+    num_features: int = 250,
+    input_url: AnyUrl | str | None = None,
+    config_url: AnyUrl | str | None = None,
+) -> str:
+    """Create and submit an Argo workflow for parallel processing of geospatial data.
+
+    This workflow follows the pattern from the 'ogdc-recipe-ice-basins-pdg' Argo workflow,
+    using Hera's Python API instead of YAML.
+
+    Args:
+        recipe_config: The recipe configuration
+        wait: Whether to wait for the workflow to complete
+        custom_image: Optional custom image to use for all containers
+        custom_tag: Optional custom tag for the image
+        custom_namespace: Optional custom namespace for the workflow
+        update_global: If True, update the global image config; if False, only apply to this workflow
+        enable_rasterize: Whether to enable the rasterize step
+        num_features: Number of features to process in each batch
+        input_url: URL to the input data file
+        config_url: URL to the visualization configuration file
+
+    Returns:
+        The name of the submitted workflow
+    """
+    with Workflow(
+        generate_name=f"{recipe_config.id}-",
+        entrypoint="main",
+        namespace="qgnet",
```

Comment:
Should this get its value from the global argo config, custom namespace, or both?